### DISCUSSION TITLE
Wrong linux version being downloaded when invoked

### DISCUSSION THREAD

**Discussion by sc6l6d3v** [flapdoodle-oss/de.flapdoodle.embed.mongo#539] [Q&A]
I'm using mongo4cats ("io.github.kirill5k" %% "mongo4cats-core" % "0.7.3") which invokes flapdoodle to download an embedded mongod instance. I am on alma linux 8.10. When I run the tests on mongo4cats that call for an embedded mongod they download the correct architecture. When I copy the same test verbatim into my Scala sbt project, it downloads the ubuntu version.

Here is the ubuntu version:

(cd ~/.embedmongo; find . -ls)
41205790 4 drwxrwxr-x 4 hkatz hkatz 4096 Aug 14 23:27 .
41205791 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 23:27 ./archives
41205796 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 23:27 ./archives/https-fastdl-mongodb-org
41205797 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 23:27 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb
41205798 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 23:27 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/linuxmongodb-linux-x86-64-ubuntu2004-5-0-14-tgz
41205799 4 drwxrwxr-x 2 hkatz hkatz 4096 Aug 14 23:27 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/linuxmongodb-linux-x86-64-ubuntu2004-5-0-14-tgz/62fa5e5dc65482463afa9081cf2bd981e67390f36b0f3d50744e200f78ce3884
41193384 84552 -rw------- 1 hkatz hkatz 86576813 Aug 14 23:27 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/linuxmongodb-linux-x86-64-ubuntu2004-5-0-14-tgz/62fa5e5dc65482463afa9081cf2bd981e67390f36b0f3d50744e200f78ce3884/archive.tgz

which doesn't work as the executable is bound to glibc_2.29 which doesn't exist on Alma

and here is the RHEL version (which works):

(cd ~/.embedmongo; find . -ls)
41205781 4 drwxrwxr-x 4 hkatz hkatz 4096 Aug 14 22:01 .
41205782 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 22:01 ./archives
41205785 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 22:01 ./archives/https-fastdl-mongodb-org
41205786 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 22:01 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb
41205787 4 drwxrwxr-x 3 hkatz hkatz 4096 Aug 14 22:01 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/linuxmongodb-linux-x86-64-rhel80-5-0-14-tgz
41205788 4 drwxrwxr-x 2 hkatz hkatz 4096 Aug 14 22:01 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/linuxmongodb-linux-x86-64-rhel80-5-0-14-tgz/f1e5f76664247e125f251668a97eefed95e9bd617290f56d346c58ddb25e391b
41169227 84556 -rw------- 1 hkatz hkatz 86581082 Aug 14 22:01 ./archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/linuxmongodb-linux-x86-64-rhel80-5-0-14-tgz/f1e5f76664247e125f251668a97eefed95e9bd617290f56d346c58ddb25e391b/archive.tgz

I am using java 21 in my repo, if that makes a difference. 

Any suggestions how to trouble shoot this incorrect choice breaking several tests that use mongo?

Thanks,
Sc6l6D3v

**Comment by michaelmosmann** [comment]
@sc6l6d3v you might try do enable some logging (see https://github.com/flapdoodle-oss/de.flapdoodle.os?tab=readme-ov-file#run) .. i guess that the os detection will make a difference.. but i still wonder why .. but maybe there is a difference how your sbt project resolves transient dependencies and .. takes a different version as the other project..  so i would suggest to use the same version for de.flapdoodle.embed.mongo ..

**Comment by sc6l6d3v** [comment]
Thanks for the suggestion. The root cause was that the release versions of flapdoodle were older than the ones in the mongo4cats repo. New ones detected my alma distro while the previous de.flapdoodle.os:1.5.3 did not add alma to the distribution list:

![image](https://github.com/user-attachments/assets/191784f2-addb-40ed-a35f-c1440478bb7c)

I cannot be ahead of mongo4cats but I will suggest a PR for him to catch up to your 1.7.2.

**Reply by michaelmosmann** [reply]
@sc6l6d3v the should use only one dependency so that there is always the matching set of all transient dependencies (its de.flapdoodle.embed.mongo).
